{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# MSigDB Abstract Wrangling\n",
    "---"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Process used to collect raw data\n",
    "\n",
    "1. Open the geneset page (ex. https://www.gsea-msigdb.org/gsea/msigdb/human/geneset/HALLMARK_APOPTOSIS.html)\n",
    "2. In \"Related gene sets\", click on (show {X} founder gene sets for this hallmark gene set)\n",
    "3. Scroll down to where it says, \"Download founder gene sets as: gmt | gmx | xml\"\n",
    "4. Download xml\n",
    "5. Upload it to a folder with the other gene sets \n",
    "\n",
    "\n",
    "This XML file will contain all the abstracts from all the founder set associated with the original gene set.\n",
    "\n",
    "**Note:** This can be done programmatically by iterating through all the pages linked on the MSigDB website, but for the scope of this study (50 genesets) it was faster to just manually download the sets."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Import necessary packages"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "pd.set_option('max_colwidth', 200)\n",
    "import numpy as np\n",
    "import os\n",
    "\n",
    "import xmltodict\n",
    "import json"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Create the dataframe and add paths to XML files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>NAME</th>\n",
       "      <th>FILENAME</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>count</th>\n",
       "      <td>50</td>\n",
       "      <td>50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>unique</th>\n",
       "      <td>50</td>\n",
       "      <td>50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>top</th>\n",
       "      <td>HALLMARK_ADIPOGENESIS</td>\n",
       "      <td>hallmark-abstracts/HALLMARK_ADIPOGENESIS_FOUND...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>freq</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                         NAME  \\\n",
       "count                      50   \n",
       "unique                     50   \n",
       "top     HALLMARK_ADIPOGENESIS   \n",
       "freq                        1   \n",
       "\n",
       "                                                 FILENAME  \n",
       "count                                                  50  \n",
       "unique                                                 50  \n",
       "top     hallmark-abstracts/HALLMARK_ADIPOGENESIS_FOUND...  \n",
       "freq                                                    1  "
      ]
     },
     "execution_count": 84,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def geneset_names_to_df(names, folder):\n",
    "    '''\n",
    "    names: text or csv file containing the names of the gene set used\n",
    "    folder: path to the folder containing the necessary XML files \n",
    "    '''\n",
    "    genesets = pd.read_csv(names)\n",
    "    genesets['FILENAME'] = abstract_folder + genesets['NAME'] + '_FOUNDERS.v2023.2.Hs.xml'\n",
    "    return genesets\n",
    "\n",
    "genesets = geneset_names_to_df('hallmark-names.txt','hallmark-abtracts/')\n",
    "genesets.describe()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Extract only the @DESCRIPTION_BRIEF and @DESCRIPTION_FULL from the XML file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "metadata": {},
   "outputs": [],
   "source": [
    "def extract_abstracts_to_dict(filename):\n",
    "    with open(filename) as file:\n",
    "        abstracts = xmltodict.parse(file.read())\n",
    "        counter = 0\n",
    "        abstract_text = {}\n",
    "        for founder in abstracts['MSIGDB']['GENESET']:\n",
    "            if founder['@DESCRIPTION_BRIEF'] or founder['@DESCRIPTION_FULL']:\n",
    "                abstract_text[counter] = f'{founder[\"@DESCRIPTION_BRIEF\"]} {founder[\"@DESCRIPTION_FULL\"]}'\n",
    "            counter += 1\n",
    "    return abstract_text\n",
    "\n",
    "genesets['ABSTRACTS'] = genesets['FILENAME'].apply(extract_abstracts_to_dict)\n",
    "genesets['ABSTRACT_COUNT'] = genesets['ABSTRACTS'].apply(lambda x: len(x))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0    {0: 'Genes down-regulated in the invasive ductal carcinoma (IDC) compared to the invasive lobular carcinoma (ILC), the two major pathological types of breast cancer. Invasive ductal carcinomas (ID...\n",
       "Name: ABSTRACTS, dtype: object"
      ]
     },
     "execution_count": 111,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genesets[genesets['NAME'] == 'HALLMARK_ADIPOGENESIS']['ABSTRACTS']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>NAME</th>\n",
       "      <th>FILENAME</th>\n",
       "      <th>ABSTRACTS</th>\n",
       "      <th>ABSTRACT_COUNT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HALLMARK_ADIPOGENESIS</td>\n",
       "      <td>hallmark-abstracts/HALLMARK_ADIPOGENESIS_FOUNDERS.v2023.2.Hs.xml</td>\n",
       "      <td>{0: 'Genes down-regulated in the invasive ductal carcinoma (IDC) compared to the invasive lobular carcinoma (ILC), the two major pathological types of breast cancer. Invasive ductal carcinomas (ID...</td>\n",
       "      <td>36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HALLMARK_ALLOGRAFT_REJECTION</td>\n",
       "      <td>hallmark-abstracts/HALLMARK_ALLOGRAFT_REJECTION_FOUNDERS.v2023.2.Hs.xml</td>\n",
       "      <td>{0: 'Genes down-regulated in cancer stem cells derived from glyoblastoma tumors: CD133+ [GeneID=8842]  vs. CD133- cells. Although glioblastomas show the same histologic phenotype, biological hallm...</td>\n",
       "      <td>189</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HALLMARK_ANDROGEN_RESPONSE</td>\n",
       "      <td>hallmark-abstracts/HALLMARK_ANDROGEN_RESPONSE_FOUNDERS.v2023.2.Hs.xml</td>\n",
       "      <td>{0: 'Genes having at least one occurence of the motif ACGCACA in their 3' untranslated region. The motif represents putative target (that is, seed match) of human mature miRNA hsa-miR-210 (v7.1 mi...</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HALLMARK_ANGIOGENESIS</td>\n",
       "      <td>hallmark-abstracts/HALLMARK_ANGIOGENESIS_FOUNDERS.v2023.2.Hs.xml</td>\n",
       "      <td>{0: 'Binding to a carbohydrate, which includes monosaccharides, oligosaccharides and polysaccharides as well as substances derived from monosaccharides by reduction of the carbonyl group (alditols...</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HALLMARK_APICAL_JUNCTION</td>\n",
       "      <td>hallmark-abstracts/HALLMARK_APICAL_JUNCTION_FOUNDERS.v2023.2.Hs.xml</td>\n",
       "      <td>{0: 'Genes annotated by the GO term GO:0051017. The assembly of actin filament bundles; actin filaments are on the same axis but may be oriented with the same or opposite polarities and may be pac...</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>HALLMARK_APICAL_SURFACE</td>\n",
       "      <td>hallmark-abstracts/HALLMARK_APICAL_SURFACE_FOUNDERS.v2023.2.Hs.xml</td>\n",
       "      <td>{0: 'Genes annotated by the GO term GO:0044463. Any constituent part of a cell projection, a prolongation or process extending from a cell, e.g. a flagellum or axon. ', 1: 'Genes in cytogenetic ba...</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>HALLMARK_APOPTOSIS</td>\n",
       "      <td>hallmark-abstracts/HALLMARK_APOPTOSIS_FOUNDERS.v2023.2.Hs.xml</td>\n",
       "      <td>{0: 'Genes up-regulated in immunoglobulin light chain amyloidosis plasma cells (ALPC) compared to multiple myeloma (MM) cells. Immunoglobulin light chain amyloidosis (AL) is characterized by a clo...</td>\n",
       "      <td>80</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>HALLMARK_BILE_ACID_METABOLISM</td>\n",
       "      <td>hallmark-abstracts/HALLMARK_BILE_ACID_METABOLISM_FOUNDERS.v2023.2.Hs.xml</td>\n",
       "      <td>{0: 'Genes in cytogenetic band chr11p Genes in cytogenetic band chr11p', 1: 'Genes in cytogenetic band chr15q Genes in cytogenetic band chr15q', 2: 'The chemical reactions and pathways involving b...</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>HALLMARK_CHOLESTEROL_HOMEOSTASIS</td>\n",
       "      <td>hallmark-abstracts/HALLMARK_CHOLESTEROL_HOMEOSTASIS_FOUNDERS.v2023.2.Hs.xml</td>\n",
       "      <td>{0: 'Genes up-regulated during prostate cancer progression in the JOCK1 model due to inducible activation of FGFR1 [GeneID=2260] gene in prostate. Fibroblast Growth Factor Receptor-1 (FGFR1) is co...</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>HALLMARK_COAGULATION</td>\n",
       "      <td>hallmark-abstracts/HALLMARK_COAGULATION_FOUNDERS.v2023.2.Hs.xml</td>\n",
       "      <td>{0: 'STAT1 [GeneID=6772] targets in hematopoetic signaling. Hematopoiesis is the cumulative result of intricately regulated signaling pathways that are mediated by cytokines and their receptors. P...</td>\n",
       "      <td>71</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>HALLMARK_COMPLEMENT</td>\n",
       "      <td>hallmark-abstracts/HALLMARK_COMPLEMENT_FOUNDERS.v2023.2.Hs.xml</td>\n",
       "      <td>{0: 'STAT1 [GeneID=6772] targets in hematopoetic signaling. Hematopoiesis is the cumulative result of intricately regulated signaling pathways that are mediated by cytokines and their receptors. P...</td>\n",
       "      <td>71</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>HALLMARK_DNA_REPAIR</td>\n",
       "      <td>hallmark-abstracts/HALLMARK_DNA_REPAIR_FOUNDERS.v2023.2.Hs.xml</td>\n",
       "      <td>{0: 'Ensembl 110 genes in cytogenetic band chr19q12 ', 1: 'Genes annotated by the GO term GO:0000428. A protein complex that possesses DNA-directed RNA polymerase activity. ', 2: 'The extension of...</td>\n",
       "      <td>44</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>HALLMARK_E2F_TARGETS</td>\n",
       "      <td>hallmark-abstracts/HALLMARK_E2F_TARGETS_FOUNDERS.v2023.2.Hs.xml</td>\n",
       "      <td>{0: 'Genes down-regulated in MEF cells (embryonic fibroblast) expressing ~25% of YY1 [GeneID=7528]. Constitutive ablation of the Yin Yang 1 (YY1) transcription factor in mice results in peri-impla...</td>\n",
       "      <td>420</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION</td>\n",
       "      <td>hallmark-abstracts/HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION_FOUNDERS.v2023.2.Hs.xml</td>\n",
       "      <td>{0: 'Invasiveness signature resulting from cancer cell/microenvironment interaction. BACKGROUND: The biological mechanisms underlying cancer cell motility and invasiveness remain unclear, although...</td>\n",
       "      <td>106</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>HALLMARK_ESTROGEN_RESPONSE_EARLY</td>\n",
       "      <td>hallmark-abstracts/HALLMARK_ESTROGEN_RESPONSE_EARLY_FOUNDERS.v2023.2.Hs.xml</td>\n",
       "      <td>{0: 'Genes up-regulated in MDA-MB-231 cells (breast cancer) after knockdown of ZEB1 [GeneID=6935] by RNAi. Epithelial to mesenchymal transition (EMT) is implicated in the progression of primary tu...</td>\n",
       "      <td>59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>HALLMARK_ESTROGEN_RESPONSE_LATE</td>\n",
       "      <td>hallmark-abstracts/HALLMARK_ESTROGEN_RESPONSE_LATE_FOUNDERS.v2023.2.Hs.xml</td>\n",
       "      <td>{0: 'Genes up-regulated in MDA-MB-231 cells (breast cancer) after knockdown of ZEB1 [GeneID=6935] by RNAi. Epithelial to mesenchymal transition (EMT) is implicated in the progression of primary tu...</td>\n",
       "      <td>59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>HALLMARK_FATTY_ACID_METABOLISM</td>\n",
       "      <td>hallmark-abstracts/HALLMARK_FATTY_ACID_METABOLISM_FOUNDERS.v2023.2.Hs.xml</td>\n",
       "      <td>{0: 'Ensembl 110 genes in cytogenetic band chr4q23 ', 1: 'Genes annotated by the GO term GO:0019395. The removal of one or more electrons from a fatty acid, with or without the concomitant removal...</td>\n",
       "      <td>53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>HALLMARK_G2M_CHECKPOINT</td>\n",
       "      <td>hallmark-abstracts/HALLMARK_G2M_CHECKPOINT_FOUNDERS.v2023.2.Hs.xml</td>\n",
       "      <td>{0: 'Genes down-regulated in MEF cells (embryonic fibroblast) expressing ~25% of YY1 [GeneID=7528]. Constitutive ablation of the Yin Yang 1 (YY1) transcription factor in mice results in peri-impla...</td>\n",
       "      <td>420</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>HALLMARK_GLYCOLYSIS</td>\n",
       "      <td>hallmark-abstracts/HALLMARK_GLYCOLYSIS_FOUNDERS.v2023.2.Hs.xml</td>\n",
       "      <td>{0: 'Feeder Pathways for Glycolysis The glycolytic pathway begins with the simple sugar glucose and leads to pyruvate and eventually the Kreb's cycle. Dietary carbohydrates include a variety of su...</td>\n",
       "      <td>87</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>HALLMARK_HEDGEHOG_SIGNALING</td>\n",
       "      <td>hallmark-abstracts/HALLMARK_HEDGEHOG_SIGNALING_FOUNDERS.v2023.2.Hs.xml</td>\n",
       "      <td>{0: 'Genes annotated by the GO term GO:0048856. The biological process whose specific outcome is the progression of an anatomical structure from an initial condition to its mature state. This proc...</td>\n",
       "      <td>79</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>HALLMARK_HEME_METABOLISM</td>\n",
       "      <td>hallmark-abstracts/HALLMARK_HEME_METABOLISM_FOUNDERS.v2023.2.Hs.xml</td>\n",
       "      <td>{0: 'Genes down-regulated in K562 cells (lymphoblast) by MYC [GeneID=4609] in the presence of CKN1B [GeneID=1027]. Inhibition of differentiation has been proposed as an important mechanism for Myc...</td>\n",
       "      <td>36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>HALLMARK_HYPOXIA</td>\n",
       "      <td>hallmark-abstracts/HALLMARK_HYPOXIA_FOUNDERS.v2023.2.Hs.xml</td>\n",
       "      <td>{0: 'Feeder Pathways for Glycolysis The glycolytic pathway begins with the simple sugar glucose and leads to pyruvate and eventually the Kreb's cycle. Dietary carbohydrates include a variety of su...</td>\n",
       "      <td>87</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>HALLMARK_IL2_STAT5_SIGNALING</td>\n",
       "      <td>hallmark-abstracts/HALLMARK_IL2_STAT5_SIGNALING_FOUNDERS.v2023.2.Hs.xml</td>\n",
       "      <td>{0: 'Genes representing a co-expression network in atopic CD4 [GeneID=920] T lymphocyte responses. Complex cellular functions within immunoinflammatory cascades are conducted by networks of intera...</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>HALLMARK_IL6_JAK_STAT3_SIGNALING</td>\n",
       "      <td>hallmark-abstracts/HALLMARK_IL6_JAK_STAT3_SIGNALING_FOUNDERS.v2023.2.Hs.xml</td>\n",
       "      <td>{0: 'IL-10 Anti-inflammatory Signaling Pathway IL-10 is a cytokine with potent anti-inflammatory properties, repressing the expression of inflammatory cytokines such as TNF-alpha, IL-6 and IL-1 by...</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>HALLMARK_INFLAMMATORY_RESPONSE</td>\n",
       "      <td>hallmark-abstracts/HALLMARK_INFLAMMATORY_RESPONSE_FOUNDERS.v2023.2.Hs.xml</td>\n",
       "      <td>{0: 'Genes changed in HBE cells (bronchial epithelium) after treatment with IL22 and IL17A [GeneID=50616;3605]. Emerging evidence supports the concept that T helper type 17 (T(H)17) cells, in addi...</td>\n",
       "      <td>120</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>HALLMARK_INTERFERON_ALPHA_RESPONSE</td>\n",
       "      <td>hallmark-abstracts/HALLMARK_INTERFERON_ALPHA_RESPONSE_FOUNDERS.v2023.2.Hs.xml</td>\n",
       "      <td>{0: 'Genes down-regulated in DAOY cells (medulloblastoma) upon knockdown of ALK [Gene ID=238] gene by RNAi. Analysis of DAOY medulloblastoma cells following RNAi knockdown of Alk.', 1: 'Genes up-r...</td>\n",
       "      <td>82</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>HALLMARK_INTERFERON_GAMMA_RESPONSE</td>\n",
       "      <td>hallmark-abstracts/HALLMARK_INTERFERON_GAMMA_RESPONSE_FOUNDERS.v2023.2.Hs.xml</td>\n",
       "      <td>{0: 'Genes down-regulated in DAOY cells (medulloblastoma) upon knockdown of ALK [Gene ID=238] gene by RNAi. Analysis of DAOY medulloblastoma cells following RNAi knockdown of Alk.', 1: 'Genes up-r...</td>\n",
       "      <td>82</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>HALLMARK_KRAS_SIGNALING_DN</td>\n",
       "      <td>hallmark-abstracts/HALLMARK_KRAS_SIGNALING_DN_FOUNDERS.v2023.2.Hs.xml</td>\n",
       "      <td>{0: 'Genes down-regulated in MCF10A cells (breast cancer) upon knockdown of BRCA1 [Gene ID=672] gene by RNAi. MCF10A cells in 3-D culture for 15 h after treatment with BRCA1-RNAi vs Luc-RNAi adeno...</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>HALLMARK_KRAS_SIGNALING_UP</td>\n",
       "      <td>hallmark-abstracts/HALLMARK_KRAS_SIGNALING_UP_FOUNDERS.v2023.2.Hs.xml</td>\n",
       "      <td>{0: 'Genes up-regulated in four lineages of epithelial cell lines over-expressing an oncogenic form of KRAS [Gene ID=3845] gene. The proto-oncogene KRAS is mutated in a wide array of human cancers...</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>HALLMARK_MITOTIC_SPINDLE</td>\n",
       "      <td>hallmark-abstracts/HALLMARK_MITOTIC_SPINDLE_FOUNDERS.v2023.2.Hs.xml</td>\n",
       "      <td>{0: 'Genes annotated by the GO term GO:0030036. A process that is carried out at the cellular level which results in the formation, arrangement of constituent parts, or disassembly of cytoskeletal...</td>\n",
       "      <td>108</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>HALLMARK_MTORC1_SIGNALING</td>\n",
       "      <td>hallmark-abstracts/HALLMARK_MTORC1_SIGNALING_FOUNDERS.v2023.2.Hs.xml</td>\n",
       "      <td>{0: 'Genes down-regulated during prostate cancer progression in the JOCK1 model due to inducible activation of FGFR1 [GeneID=2260] gene in prostate. Fibroblast Growth Factor Receptor-1 (FGFR1) is ...</td>\n",
       "      <td>484</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>HALLMARK_MYC_TARGETS_V1</td>\n",
       "      <td>hallmark-abstracts/HALLMARK_MYC_TARGETS_V1_FOUNDERS.v2023.2.Hs.xml</td>\n",
       "      <td>{0: 'Proteins interacting with AIRE [GeneID=326], based on massspectroscopy analysis of co-immunoprecipitates in 293T cells (embryonic kidney). Aire induces the expression of a battery of peripher...</td>\n",
       "      <td>404</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>HALLMARK_MYC_TARGETS_V2</td>\n",
       "      <td>hallmark-abstracts/HALLMARK_MYC_TARGETS_V2_FOUNDERS.v2023.2.Hs.xml</td>\n",
       "      <td>{0: 'Genes selected in supervised analyses to discriminate cells expressing c-Myc [GeneID=4609] from control cells expressing GFP. The development of an oncogenic state is a complex process involv...</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>HALLMARK_MYOGENESIS</td>\n",
       "      <td>hallmark-abstracts/HALLMARK_MYOGENESIS_FOUNDERS.v2023.2.Hs.xml</td>\n",
       "      <td>{0: 'Genes down-regulated in embryonic stem cells with BRCA1 [GeneID=672] loss of function (LOF). BRCA1 gene is a tumor suppressor for breast and ovarian cancers with the putative role in DNA repa...</td>\n",
       "      <td>63</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>HALLMARK_NOTCH_SIGNALING</td>\n",
       "      <td>hallmark-abstracts/HALLMARK_NOTCH_SIGNALING_FOUNDERS.v2023.2.Hs.xml</td>\n",
       "      <td>{0: 'Inactivation of Gsk3 by AKT causes accumulation of b-catenin in Alveolar Macrophages Lipopolysaccharide. One of the key actions of AKT is to block apoptosis. AKT phosphorylation of NF-kB prom...</td>\n",
       "      <td>49</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>HALLMARK_OXIDATIVE_PHOSPHORYLATION</td>\n",
       "      <td>hallmark-abstracts/HALLMARK_OXIDATIVE_PHOSPHORYLATION_FOUNDERS.v2023.2.Hs.xml</td>\n",
       "      <td>{0: 'Genes down-regulated in A2780 cells (ovarian carcinoma) exposed to fenretinide [PubChem=1744]. Fenretinide (4-HPR) is a synthetic retinoid with antitumor activity, which induces apoptosis in ...</td>\n",
       "      <td>93</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>HALLMARK_P53_PATHWAY</td>\n",
       "      <td>hallmark-abstracts/HALLMARK_P53_PATHWAY_FOUNDERS.v2023.2.Hs.xml</td>\n",
       "      <td>{0: 'Genes up-regulated in MEF cells (embryonic fibroblast) expressing ~25% of YY1 [GeneID=7528]. Constitutive ablation of the Yin Yang 1 (YY1) transcription factor in mice results in peri-implant...</td>\n",
       "      <td>85</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>HALLMARK_PANCREAS_BETA_CELLS</td>\n",
       "      <td>hallmark-abstracts/HALLMARK_PANCREAS_BETA_CELLS_FOUNDERS.v2023.2.Hs.xml</td>\n",
       "      <td>{0: 'Set 'Core 9': 'embryonic stem cell' transcription regulators that are preferentially and coordinately overexpressed in the high-grade, ER-negative breast cancer tumors. Cancer cells possess t...</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>HALLMARK_PEROXISOME</td>\n",
       "      <td>hallmark-abstracts/HALLMARK_PEROXISOME_FOUNDERS.v2023.2.Hs.xml</td>\n",
       "      <td>{0: 'Genes in cytogenetic band chr11p Genes in cytogenetic band chr11p', 1: 'Genes in cytogenetic band chr15q Genes in cytogenetic band chr15q', 2: 'The chemical reactions and pathways involving b...</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>HALLMARK_PI3K_AKT_MTOR_SIGNALING</td>\n",
       "      <td>hallmark-abstracts/HALLMARK_PI3K_AKT_MTOR_SIGNALING_FOUNDERS.v2023.2.Hs.xml</td>\n",
       "      <td>{0: 'Major ELAVL4 [GeneID=1996] associated mRNAs encoding proteins with functions in neuronal physiology. Neuronal development and plasticity are maintained by tightly regulated gene expression pr...</td>\n",
       "      <td>591</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>HALLMARK_PROTEIN_SECRETION</td>\n",
       "      <td>hallmark-abstracts/HALLMARK_PROTEIN_SECRETION_FOUNDERS.v2023.2.Hs.xml</td>\n",
       "      <td>{0: 'ADP-Ribosylation Factor ADP-ribosylation factors (ARFs) are 20-kDa guanine nucleotide-binding proteins, members of the Ras GTPase superfamily that were initially recognized and purified becau...</td>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY</td>\n",
       "      <td>hallmark-abstracts/HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY_FOUNDERS.v2023.2.Hs.xml</td>\n",
       "      <td>{0: 'Genes up-regulated in GH3 cells (pituitary cancer) after treatment with LIF [GeneID=3976]. Leukemia inhibitory factor (LIF) mediates the hypothalamo-pituitary-adrenal stress response. Transge...</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>HALLMARK_SPERMATOGENESIS</td>\n",
       "      <td>hallmark-abstracts/HALLMARK_SPERMATOGENESIS_FOUNDERS.v2023.2.Hs.xml</td>\n",
       "      <td>{0: 'Neighborhood of CCNA1 Neighborhood of CCNA1 cyclin A1  in the GNF2 expression compendium', 1: 'Neighborhood of MLF1 Neighborhood of MLF1 myeloid leukemia factor 1  in the GNF2 expression comp...</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>HALLMARK_TGF_BETA_SIGNALING</td>\n",
       "      <td>hallmark-abstracts/HALLMARK_TGF_BETA_SIGNALING_FOUNDERS.v2023.2.Hs.xml</td>\n",
       "      <td>{0: 'ALK in cardiac myocytes Heart formation is cued by a combination of positive and negative signals from surrounding tissues. Inhibitory signals that block heart formation in anterior paraxial ...</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>HALLMARK_TNFA_SIGNALING_VIA_NFKB</td>\n",
       "      <td>hallmark-abstracts/HALLMARK_TNFA_SIGNALING_VIA_NFKB_FOUNDERS.v2023.2.Hs.xml</td>\n",
       "      <td>{0: 'Genes most profoundly induced in HUVEC cells (endothelium) by VEGFA [GeneID=7422]. Vascular endothelial growth factor (VEGF) is one of the most important factors that stimulate angiogenesis a...</td>\n",
       "      <td>132</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>HALLMARK_UNFOLDED_PROTEIN_RESPONSE</td>\n",
       "      <td>hallmark-abstracts/HALLMARK_UNFOLDED_PROTEIN_RESPONSE_FOUNDERS.v2023.2.Hs.xml</td>\n",
       "      <td>{0: 'Ensembl 110 genes in cytogenetic band chr13q21 ', 1: 'Genes from the red module which are up-regulated in HAEC cells (primary aortic endothelium) after exposure to the oxidized 1-palmitoyl-2-...</td>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>HALLMARK_UV_RESPONSE_DN</td>\n",
       "      <td>hallmark-abstracts/HALLMARK_UV_RESPONSE_DN_FOUNDERS.v2023.2.Hs.xml</td>\n",
       "      <td>{0: 'Genes down-regulated in primary fibroblast cell culture after infection with HCMV (AD169 strain) at 10 h time point that were not down-regulated at the previous time point, 8 h. The effect of...</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>HALLMARK_UV_RESPONSE_UP</td>\n",
       "      <td>hallmark-abstracts/HALLMARK_UV_RESPONSE_UP_FOUNDERS.v2023.2.Hs.xml</td>\n",
       "      <td>{0: 'Genes up-regulated in primary fibroblast cell culture after infection with HCMV (AD169 strain) at 10 h time point that were not up-regulated at the previous time point, 8 h. The effect of hum...</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>HALLMARK_WNT_BETA_CATENIN_SIGNALING</td>\n",
       "      <td>hallmark-abstracts/HALLMARK_WNT_BETA_CATENIN_SIGNALING_FOUNDERS.v2023.2.Hs.xml</td>\n",
       "      <td>{0: 'Inactivation of Gsk3 by AKT causes accumulation of b-catenin in Alveolar Macrophages Lipopolysaccharide. One of the key actions of AKT is to block apoptosis. AKT phosphorylation of NF-kB prom...</td>\n",
       "      <td>49</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>HALLMARK_XENOBIOTIC_METABOLISM</td>\n",
       "      <td>hallmark-abstracts/HALLMARK_XENOBIOTIC_METABOLISM_FOUNDERS.v2023.2.Hs.xml</td>\n",
       "      <td>{0: 'Genes down-regulated in hepatocellular carcinoma (HCC) compared to normal liver samples. There is widespread interest in efficient characterization of differences between tumor and normal sam...</td>\n",
       "      <td>124</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                          NAME  \\\n",
       "0                        HALLMARK_ADIPOGENESIS   \n",
       "1                 HALLMARK_ALLOGRAFT_REJECTION   \n",
       "2                   HALLMARK_ANDROGEN_RESPONSE   \n",
       "3                        HALLMARK_ANGIOGENESIS   \n",
       "4                     HALLMARK_APICAL_JUNCTION   \n",
       "5                      HALLMARK_APICAL_SURFACE   \n",
       "6                           HALLMARK_APOPTOSIS   \n",
       "7                HALLMARK_BILE_ACID_METABOLISM   \n",
       "8             HALLMARK_CHOLESTEROL_HOMEOSTASIS   \n",
       "9                         HALLMARK_COAGULATION   \n",
       "10                         HALLMARK_COMPLEMENT   \n",
       "11                         HALLMARK_DNA_REPAIR   \n",
       "12                        HALLMARK_E2F_TARGETS   \n",
       "13  HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION   \n",
       "14            HALLMARK_ESTROGEN_RESPONSE_EARLY   \n",
       "15             HALLMARK_ESTROGEN_RESPONSE_LATE   \n",
       "16              HALLMARK_FATTY_ACID_METABOLISM   \n",
       "17                     HALLMARK_G2M_CHECKPOINT   \n",
       "18                         HALLMARK_GLYCOLYSIS   \n",
       "19                 HALLMARK_HEDGEHOG_SIGNALING   \n",
       "20                    HALLMARK_HEME_METABOLISM   \n",
       "21                            HALLMARK_HYPOXIA   \n",
       "22                HALLMARK_IL2_STAT5_SIGNALING   \n",
       "23            HALLMARK_IL6_JAK_STAT3_SIGNALING   \n",
       "24              HALLMARK_INFLAMMATORY_RESPONSE   \n",
       "25          HALLMARK_INTERFERON_ALPHA_RESPONSE   \n",
       "26          HALLMARK_INTERFERON_GAMMA_RESPONSE   \n",
       "27                  HALLMARK_KRAS_SIGNALING_DN   \n",
       "28                  HALLMARK_KRAS_SIGNALING_UP   \n",
       "29                    HALLMARK_MITOTIC_SPINDLE   \n",
       "30                   HALLMARK_MTORC1_SIGNALING   \n",
       "31                     HALLMARK_MYC_TARGETS_V1   \n",
       "32                     HALLMARK_MYC_TARGETS_V2   \n",
       "33                         HALLMARK_MYOGENESIS   \n",
       "34                    HALLMARK_NOTCH_SIGNALING   \n",
       "35          HALLMARK_OXIDATIVE_PHOSPHORYLATION   \n",
       "36                        HALLMARK_P53_PATHWAY   \n",
       "37                HALLMARK_PANCREAS_BETA_CELLS   \n",
       "38                         HALLMARK_PEROXISOME   \n",
       "39            HALLMARK_PI3K_AKT_MTOR_SIGNALING   \n",
       "40                  HALLMARK_PROTEIN_SECRETION   \n",
       "41    HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY   \n",
       "42                    HALLMARK_SPERMATOGENESIS   \n",
       "43                 HALLMARK_TGF_BETA_SIGNALING   \n",
       "44            HALLMARK_TNFA_SIGNALING_VIA_NFKB   \n",
       "45          HALLMARK_UNFOLDED_PROTEIN_RESPONSE   \n",
       "46                     HALLMARK_UV_RESPONSE_DN   \n",
       "47                     HALLMARK_UV_RESPONSE_UP   \n",
       "48         HALLMARK_WNT_BETA_CATENIN_SIGNALING   \n",
       "49              HALLMARK_XENOBIOTIC_METABOLISM   \n",
       "\n",
       "                                                                                 FILENAME  \\\n",
       "0                        hallmark-abstracts/HALLMARK_ADIPOGENESIS_FOUNDERS.v2023.2.Hs.xml   \n",
       "1                 hallmark-abstracts/HALLMARK_ALLOGRAFT_REJECTION_FOUNDERS.v2023.2.Hs.xml   \n",
       "2                   hallmark-abstracts/HALLMARK_ANDROGEN_RESPONSE_FOUNDERS.v2023.2.Hs.xml   \n",
       "3                        hallmark-abstracts/HALLMARK_ANGIOGENESIS_FOUNDERS.v2023.2.Hs.xml   \n",
       "4                     hallmark-abstracts/HALLMARK_APICAL_JUNCTION_FOUNDERS.v2023.2.Hs.xml   \n",
       "5                      hallmark-abstracts/HALLMARK_APICAL_SURFACE_FOUNDERS.v2023.2.Hs.xml   \n",
       "6                           hallmark-abstracts/HALLMARK_APOPTOSIS_FOUNDERS.v2023.2.Hs.xml   \n",
       "7                hallmark-abstracts/HALLMARK_BILE_ACID_METABOLISM_FOUNDERS.v2023.2.Hs.xml   \n",
       "8             hallmark-abstracts/HALLMARK_CHOLESTEROL_HOMEOSTASIS_FOUNDERS.v2023.2.Hs.xml   \n",
       "9                         hallmark-abstracts/HALLMARK_COAGULATION_FOUNDERS.v2023.2.Hs.xml   \n",
       "10                         hallmark-abstracts/HALLMARK_COMPLEMENT_FOUNDERS.v2023.2.Hs.xml   \n",
       "11                         hallmark-abstracts/HALLMARK_DNA_REPAIR_FOUNDERS.v2023.2.Hs.xml   \n",
       "12                        hallmark-abstracts/HALLMARK_E2F_TARGETS_FOUNDERS.v2023.2.Hs.xml   \n",
       "13  hallmark-abstracts/HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION_FOUNDERS.v2023.2.Hs.xml   \n",
       "14            hallmark-abstracts/HALLMARK_ESTROGEN_RESPONSE_EARLY_FOUNDERS.v2023.2.Hs.xml   \n",
       "15             hallmark-abstracts/HALLMARK_ESTROGEN_RESPONSE_LATE_FOUNDERS.v2023.2.Hs.xml   \n",
       "16              hallmark-abstracts/HALLMARK_FATTY_ACID_METABOLISM_FOUNDERS.v2023.2.Hs.xml   \n",
       "17                     hallmark-abstracts/HALLMARK_G2M_CHECKPOINT_FOUNDERS.v2023.2.Hs.xml   \n",
       "18                         hallmark-abstracts/HALLMARK_GLYCOLYSIS_FOUNDERS.v2023.2.Hs.xml   \n",
       "19                 hallmark-abstracts/HALLMARK_HEDGEHOG_SIGNALING_FOUNDERS.v2023.2.Hs.xml   \n",
       "20                    hallmark-abstracts/HALLMARK_HEME_METABOLISM_FOUNDERS.v2023.2.Hs.xml   \n",
       "21                            hallmark-abstracts/HALLMARK_HYPOXIA_FOUNDERS.v2023.2.Hs.xml   \n",
       "22                hallmark-abstracts/HALLMARK_IL2_STAT5_SIGNALING_FOUNDERS.v2023.2.Hs.xml   \n",
       "23            hallmark-abstracts/HALLMARK_IL6_JAK_STAT3_SIGNALING_FOUNDERS.v2023.2.Hs.xml   \n",
       "24              hallmark-abstracts/HALLMARK_INFLAMMATORY_RESPONSE_FOUNDERS.v2023.2.Hs.xml   \n",
       "25          hallmark-abstracts/HALLMARK_INTERFERON_ALPHA_RESPONSE_FOUNDERS.v2023.2.Hs.xml   \n",
       "26          hallmark-abstracts/HALLMARK_INTERFERON_GAMMA_RESPONSE_FOUNDERS.v2023.2.Hs.xml   \n",
       "27                  hallmark-abstracts/HALLMARK_KRAS_SIGNALING_DN_FOUNDERS.v2023.2.Hs.xml   \n",
       "28                  hallmark-abstracts/HALLMARK_KRAS_SIGNALING_UP_FOUNDERS.v2023.2.Hs.xml   \n",
       "29                    hallmark-abstracts/HALLMARK_MITOTIC_SPINDLE_FOUNDERS.v2023.2.Hs.xml   \n",
       "30                   hallmark-abstracts/HALLMARK_MTORC1_SIGNALING_FOUNDERS.v2023.2.Hs.xml   \n",
       "31                     hallmark-abstracts/HALLMARK_MYC_TARGETS_V1_FOUNDERS.v2023.2.Hs.xml   \n",
       "32                     hallmark-abstracts/HALLMARK_MYC_TARGETS_V2_FOUNDERS.v2023.2.Hs.xml   \n",
       "33                         hallmark-abstracts/HALLMARK_MYOGENESIS_FOUNDERS.v2023.2.Hs.xml   \n",
       "34                    hallmark-abstracts/HALLMARK_NOTCH_SIGNALING_FOUNDERS.v2023.2.Hs.xml   \n",
       "35          hallmark-abstracts/HALLMARK_OXIDATIVE_PHOSPHORYLATION_FOUNDERS.v2023.2.Hs.xml   \n",
       "36                        hallmark-abstracts/HALLMARK_P53_PATHWAY_FOUNDERS.v2023.2.Hs.xml   \n",
       "37                hallmark-abstracts/HALLMARK_PANCREAS_BETA_CELLS_FOUNDERS.v2023.2.Hs.xml   \n",
       "38                         hallmark-abstracts/HALLMARK_PEROXISOME_FOUNDERS.v2023.2.Hs.xml   \n",
       "39            hallmark-abstracts/HALLMARK_PI3K_AKT_MTOR_SIGNALING_FOUNDERS.v2023.2.Hs.xml   \n",
       "40                  hallmark-abstracts/HALLMARK_PROTEIN_SECRETION_FOUNDERS.v2023.2.Hs.xml   \n",
       "41    hallmark-abstracts/HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY_FOUNDERS.v2023.2.Hs.xml   \n",
       "42                    hallmark-abstracts/HALLMARK_SPERMATOGENESIS_FOUNDERS.v2023.2.Hs.xml   \n",
       "43                 hallmark-abstracts/HALLMARK_TGF_BETA_SIGNALING_FOUNDERS.v2023.2.Hs.xml   \n",
       "44            hallmark-abstracts/HALLMARK_TNFA_SIGNALING_VIA_NFKB_FOUNDERS.v2023.2.Hs.xml   \n",
       "45          hallmark-abstracts/HALLMARK_UNFOLDED_PROTEIN_RESPONSE_FOUNDERS.v2023.2.Hs.xml   \n",
       "46                     hallmark-abstracts/HALLMARK_UV_RESPONSE_DN_FOUNDERS.v2023.2.Hs.xml   \n",
       "47                     hallmark-abstracts/HALLMARK_UV_RESPONSE_UP_FOUNDERS.v2023.2.Hs.xml   \n",
       "48         hallmark-abstracts/HALLMARK_WNT_BETA_CATENIN_SIGNALING_FOUNDERS.v2023.2.Hs.xml   \n",
       "49              hallmark-abstracts/HALLMARK_XENOBIOTIC_METABOLISM_FOUNDERS.v2023.2.Hs.xml   \n",
       "\n",
       "                                                                                                                                                                                                  ABSTRACTS  \\\n",
       "0   {0: 'Genes down-regulated in the invasive ductal carcinoma (IDC) compared to the invasive lobular carcinoma (ILC), the two major pathological types of breast cancer. Invasive ductal carcinomas (ID...   \n",
       "1   {0: 'Genes down-regulated in cancer stem cells derived from glyoblastoma tumors: CD133+ [GeneID=8842]  vs. CD133- cells. Although glioblastomas show the same histologic phenotype, biological hallm...   \n",
       "2   {0: 'Genes having at least one occurence of the motif ACGCACA in their 3' untranslated region. The motif represents putative target (that is, seed match) of human mature miRNA hsa-miR-210 (v7.1 mi...   \n",
       "3   {0: 'Binding to a carbohydrate, which includes monosaccharides, oligosaccharides and polysaccharides as well as substances derived from monosaccharides by reduction of the carbonyl group (alditols...   \n",
       "4   {0: 'Genes annotated by the GO term GO:0051017. The assembly of actin filament bundles; actin filaments are on the same axis but may be oriented with the same or opposite polarities and may be pac...   \n",
       "5   {0: 'Genes annotated by the GO term GO:0044463. Any constituent part of a cell projection, a prolongation or process extending from a cell, e.g. a flagellum or axon. ', 1: 'Genes in cytogenetic ba...   \n",
       "6   {0: 'Genes up-regulated in immunoglobulin light chain amyloidosis plasma cells (ALPC) compared to multiple myeloma (MM) cells. Immunoglobulin light chain amyloidosis (AL) is characterized by a clo...   \n",
       "7   {0: 'Genes in cytogenetic band chr11p Genes in cytogenetic band chr11p', 1: 'Genes in cytogenetic band chr15q Genes in cytogenetic band chr15q', 2: 'The chemical reactions and pathways involving b...   \n",
       "8   {0: 'Genes up-regulated during prostate cancer progression in the JOCK1 model due to inducible activation of FGFR1 [GeneID=2260] gene in prostate. Fibroblast Growth Factor Receptor-1 (FGFR1) is co...   \n",
       "9   {0: 'STAT1 [GeneID=6772] targets in hematopoetic signaling. Hematopoiesis is the cumulative result of intricately regulated signaling pathways that are mediated by cytokines and their receptors. P...   \n",
       "10  {0: 'STAT1 [GeneID=6772] targets in hematopoetic signaling. Hematopoiesis is the cumulative result of intricately regulated signaling pathways that are mediated by cytokines and their receptors. P...   \n",
       "11  {0: 'Ensembl 110 genes in cytogenetic band chr19q12 ', 1: 'Genes annotated by the GO term GO:0000428. A protein complex that possesses DNA-directed RNA polymerase activity. ', 2: 'The extension of...   \n",
       "12  {0: 'Genes down-regulated in MEF cells (embryonic fibroblast) expressing ~25% of YY1 [GeneID=7528]. Constitutive ablation of the Yin Yang 1 (YY1) transcription factor in mice results in peri-impla...   \n",
       "13  {0: 'Invasiveness signature resulting from cancer cell/microenvironment interaction. BACKGROUND: The biological mechanisms underlying cancer cell motility and invasiveness remain unclear, although...   \n",
       "14  {0: 'Genes up-regulated in MDA-MB-231 cells (breast cancer) after knockdown of ZEB1 [GeneID=6935] by RNAi. Epithelial to mesenchymal transition (EMT) is implicated in the progression of primary tu...   \n",
       "15  {0: 'Genes up-regulated in MDA-MB-231 cells (breast cancer) after knockdown of ZEB1 [GeneID=6935] by RNAi. Epithelial to mesenchymal transition (EMT) is implicated in the progression of primary tu...   \n",
       "16  {0: 'Ensembl 110 genes in cytogenetic band chr4q23 ', 1: 'Genes annotated by the GO term GO:0019395. The removal of one or more electrons from a fatty acid, with or without the concomitant removal...   \n",
       "17  {0: 'Genes down-regulated in MEF cells (embryonic fibroblast) expressing ~25% of YY1 [GeneID=7528]. Constitutive ablation of the Yin Yang 1 (YY1) transcription factor in mice results in peri-impla...   \n",
       "18  {0: 'Feeder Pathways for Glycolysis The glycolytic pathway begins with the simple sugar glucose and leads to pyruvate and eventually the Kreb's cycle. Dietary carbohydrates include a variety of su...   \n",
       "19  {0: 'Genes annotated by the GO term GO:0048856. The biological process whose specific outcome is the progression of an anatomical structure from an initial condition to its mature state. This proc...   \n",
       "20  {0: 'Genes down-regulated in K562 cells (lymphoblast) by MYC [GeneID=4609] in the presence of CKN1B [GeneID=1027]. Inhibition of differentiation has been proposed as an important mechanism for Myc...   \n",
       "21  {0: 'Feeder Pathways for Glycolysis The glycolytic pathway begins with the simple sugar glucose and leads to pyruvate and eventually the Kreb's cycle. Dietary carbohydrates include a variety of su...   \n",
       "22  {0: 'Genes representing a co-expression network in atopic CD4 [GeneID=920] T lymphocyte responses. Complex cellular functions within immunoinflammatory cascades are conducted by networks of intera...   \n",
       "23  {0: 'IL-10 Anti-inflammatory Signaling Pathway IL-10 is a cytokine with potent anti-inflammatory properties, repressing the expression of inflammatory cytokines such as TNF-alpha, IL-6 and IL-1 by...   \n",
       "24  {0: 'Genes changed in HBE cells (bronchial epithelium) after treatment with IL22 and IL17A [GeneID=50616;3605]. Emerging evidence supports the concept that T helper type 17 (T(H)17) cells, in addi...   \n",
       "25  {0: 'Genes down-regulated in DAOY cells (medulloblastoma) upon knockdown of ALK [Gene ID=238] gene by RNAi. Analysis of DAOY medulloblastoma cells following RNAi knockdown of Alk.', 1: 'Genes up-r...   \n",
       "26  {0: 'Genes down-regulated in DAOY cells (medulloblastoma) upon knockdown of ALK [Gene ID=238] gene by RNAi. Analysis of DAOY medulloblastoma cells following RNAi knockdown of Alk.', 1: 'Genes up-r...   \n",
       "27  {0: 'Genes down-regulated in MCF10A cells (breast cancer) upon knockdown of BRCA1 [Gene ID=672] gene by RNAi. MCF10A cells in 3-D culture for 15 h after treatment with BRCA1-RNAi vs Luc-RNAi adeno...   \n",
       "28  {0: 'Genes up-regulated in four lineages of epithelial cell lines over-expressing an oncogenic form of KRAS [Gene ID=3845] gene. The proto-oncogene KRAS is mutated in a wide array of human cancers...   \n",
       "29  {0: 'Genes annotated by the GO term GO:0030036. A process that is carried out at the cellular level which results in the formation, arrangement of constituent parts, or disassembly of cytoskeletal...   \n",
       "30  {0: 'Genes down-regulated during prostate cancer progression in the JOCK1 model due to inducible activation of FGFR1 [GeneID=2260] gene in prostate. Fibroblast Growth Factor Receptor-1 (FGFR1) is ...   \n",
       "31  {0: 'Proteins interacting with AIRE [GeneID=326], based on massspectroscopy analysis of co-immunoprecipitates in 293T cells (embryonic kidney). Aire induces the expression of a battery of peripher...   \n",
       "32  {0: 'Genes selected in supervised analyses to discriminate cells expressing c-Myc [GeneID=4609] from control cells expressing GFP. The development of an oncogenic state is a complex process involv...   \n",
       "33  {0: 'Genes down-regulated in embryonic stem cells with BRCA1 [GeneID=672] loss of function (LOF). BRCA1 gene is a tumor suppressor for breast and ovarian cancers with the putative role in DNA repa...   \n",
       "34  {0: 'Inactivation of Gsk3 by AKT causes accumulation of b-catenin in Alveolar Macrophages Lipopolysaccharide. One of the key actions of AKT is to block apoptosis. AKT phosphorylation of NF-kB prom...   \n",
       "35  {0: 'Genes down-regulated in A2780 cells (ovarian carcinoma) exposed to fenretinide [PubChem=1744]. Fenretinide (4-HPR) is a synthetic retinoid with antitumor activity, which induces apoptosis in ...   \n",
       "36  {0: 'Genes up-regulated in MEF cells (embryonic fibroblast) expressing ~25% of YY1 [GeneID=7528]. Constitutive ablation of the Yin Yang 1 (YY1) transcription factor in mice results in peri-implant...   \n",
       "37  {0: 'Set 'Core 9': 'embryonic stem cell' transcription regulators that are preferentially and coordinately overexpressed in the high-grade, ER-negative breast cancer tumors. Cancer cells possess t...   \n",
       "38  {0: 'Genes in cytogenetic band chr11p Genes in cytogenetic band chr11p', 1: 'Genes in cytogenetic band chr15q Genes in cytogenetic band chr15q', 2: 'The chemical reactions and pathways involving b...   \n",
       "39  {0: 'Major ELAVL4 [GeneID=1996] associated mRNAs encoding proteins with functions in neuronal physiology. Neuronal development and plasticity are maintained by tightly regulated gene expression pr...   \n",
       "40  {0: 'ADP-Ribosylation Factor ADP-ribosylation factors (ARFs) are 20-kDa guanine nucleotide-binding proteins, members of the Ras GTPase superfamily that were initially recognized and purified becau...   \n",
       "41  {0: 'Genes up-regulated in GH3 cells (pituitary cancer) after treatment with LIF [GeneID=3976]. Leukemia inhibitory factor (LIF) mediates the hypothalamo-pituitary-adrenal stress response. Transge...   \n",
       "42  {0: 'Neighborhood of CCNA1 Neighborhood of CCNA1 cyclin A1  in the GNF2 expression compendium', 1: 'Neighborhood of MLF1 Neighborhood of MLF1 myeloid leukemia factor 1  in the GNF2 expression comp...   \n",
       "43  {0: 'ALK in cardiac myocytes Heart formation is cued by a combination of positive and negative signals from surrounding tissues. Inhibitory signals that block heart formation in anterior paraxial ...   \n",
       "44  {0: 'Genes most profoundly induced in HUVEC cells (endothelium) by VEGFA [GeneID=7422]. Vascular endothelial growth factor (VEGF) is one of the most important factors that stimulate angiogenesis a...   \n",
       "45  {0: 'Ensembl 110 genes in cytogenetic band chr13q21 ', 1: 'Genes from the red module which are up-regulated in HAEC cells (primary aortic endothelium) after exposure to the oxidized 1-palmitoyl-2-...   \n",
       "46  {0: 'Genes down-regulated in primary fibroblast cell culture after infection with HCMV (AD169 strain) at 10 h time point that were not down-regulated at the previous time point, 8 h. The effect of...   \n",
       "47  {0: 'Genes up-regulated in primary fibroblast cell culture after infection with HCMV (AD169 strain) at 10 h time point that were not up-regulated at the previous time point, 8 h. The effect of hum...   \n",
       "48  {0: 'Inactivation of Gsk3 by AKT causes accumulation of b-catenin in Alveolar Macrophages Lipopolysaccharide. One of the key actions of AKT is to block apoptosis. AKT phosphorylation of NF-kB prom...   \n",
       "49  {0: 'Genes down-regulated in hepatocellular carcinoma (HCC) compared to normal liver samples. There is widespread interest in efficient characterization of differences between tumor and normal sam...   \n",
       "\n",
       "    ABSTRACT_COUNT  \n",
       "0               36  \n",
       "1              189  \n",
       "2                8  \n",
       "3               14  \n",
       "4               37  \n",
       "5               12  \n",
       "6               80  \n",
       "7               28  \n",
       "8               28  \n",
       "9               71  \n",
       "10              71  \n",
       "11              44  \n",
       "12             420  \n",
       "13             106  \n",
       "14              59  \n",
       "15              59  \n",
       "16              53  \n",
       "17             420  \n",
       "18              87  \n",
       "19              79  \n",
       "20              36  \n",
       "21              87  \n",
       "22              13  \n",
       "23              23  \n",
       "24             120  \n",
       "25              82  \n",
       "26              82  \n",
       "27              16  \n",
       "28              14  \n",
       "29             108  \n",
       "30             484  \n",
       "31             404  \n",
       "32               6  \n",
       "33              63  \n",
       "34              49  \n",
       "35              93  \n",
       "36              85  \n",
       "37              24  \n",
       "38              28  \n",
       "39             591  \n",
       "40              74  \n",
       "41              13  \n",
       "42              24  \n",
       "43              29  \n",
       "44             132  \n",
       "45              22  \n",
       "46              17  \n",
       "47              16  \n",
       "48              49  \n",
       "49             124  "
      ]
     },
     "execution_count": 123,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genesets"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
